angular2-jwt.ts

```diff
@@ -15,46 +15,41 @@ export interface IAuthConfig {
   noTokenScheme?:boolean;
 }
 
+class AuthConfigConsts {
+    public static readonly DEFAULT_TOKEN_NAME = 'id_token';
+    public static readonly DEFAULT_HEADER_NAME = 'Authorization';
+    public static readonly HEADER_PREFIX_BEARER = 'Bearer ';
+}
+
 /**
  * Sets up the authentication configuration.
  */
 
 export class AuthConfig {
-  
-  headerName: string;
-  headerPrefix: string;
-  tokenName: string;
-  tokenGetter: any;
-  noJwtError: boolean;
-  noTokenScheme: boolean;
-  globalHeaders: Array<Object>;
 
-  constructor(config:any={}) {
-    this.headerName = config.headerName || 'Authorization';
-    if (config.headerPrefix) {
-      this.headerPrefix = config.headerPrefix + ' ';
-    } else if (config.noTokenScheme) {
-      this.headerPrefix = '';
+  private _config : IAuthConfig;
+
+  constructor(config:IAuthConfig = {
```

Comment:
I don't see what's complicated about it. It is just one more interface... :confused: 

Of course your solution works, too, but it fails the point I'm trying to achieve with using interfaces, which is **type information**. I mean, we are using typescript here, so why not using the type info where possible?
The signature `AuthHttp(config:any)` does not say _anything_ about the form of the config object, so IMHO type info would significantly increase usability.

Do you have other ideas of providing type info to the config object?